scripts/truffle_project.py

```diff
@@ -55,14 +55,20 @@ def _merge_compiled_files(self, compilation_output):
         for entry in os.scandir(self.build_dir):
             if entry.is_file() and entry.name.endswith(".json") and\
                     entry.name != "Migrations.json":
-                with open(entry) as file:
-                    data = json.load(file)
-                contract_name = data["sourcePath"] + ":" + data["contractName"]
+                with open(entry) as f:
+                    data = json.load(f)
+                contract_path = data["sourcePath"]
+                contract_name = data["contractName"]
                 # check if library contract
-                if not pathlib.Path(contract_name).is_file():
-                    contract_name = os.path.join(
-                        utils.find_node_modules_dir(self.project_root),
-                        contract_name)
+                if not pathlib.Path(contract_path).is_file():
+                    node_modules_dir = utils.find_node_modules_dir(
+                        self.project_root)
+                    if node_modules_dir is None:
+                        print("[!] Error: Couldn't find a source for "
+                              f"{contract_name}")
+                        sys.exit(1)
+                    contract_path = os.path.join(node_modules_dir,
```

Comment:
Should we add a check for this, so that we can give a proper warning in case a dependency cannot be resolved?